app/models/parameter.rb

```diff
@@ -10,6 +10,29 @@ class Parameter < ActiveRecord::Base
   validates :name, :presence => true, :no_whitespace => true
 
   scoped_search :on => :name, :complete_value => true
+  scoped_search :on => :type, :complete_value => true
+  scoped_search :on => :value, :complete_value => true
+
+  # children associations must be defined here, otherwise scoped search definitions won't find them
+  belongs_to :domain, :foreign_key => :reference_id, :inverse_of => :domain_parameters
```

Comment:
Unfortunately, it does not work, if we define associations later, children don't seem to inherit them correctly in AR. If you want to experiment further, here's the gist with patch and error I experience https://gist.github.com/ares/ab2ceb70d7873ba962039831afe7007b